src/fun/scala/kinesis/mock/KCLTests.scala

```diff
@@ -139,29 +138,45 @@ class KCLTests extends munit.CatsEffectSuite with AwsFunctionalTests {
       .join(RetryPolicies.constantDelay(1.second))
     gotAllRecords <- retryingOnFailures[Boolean](
       policy,
-      identity,
-      noop[IO, Boolean]
+      IO.pure,
+      { case (_, status) =>
+        IO(
+          println(
+            s"Results queue is not full, retrying. Retry Status: ${status.toString}"
+          )
+        )
+      }
     )(
-      resources.resultsQueue.getSize.map(_ == 5)
+      resources.resultsQueue.size.map(_ == 5)
     )
-    resRecords <- resources.resultsQueue.dequeueChunk1(5).map(_.toVector)
+    resRecords <- for {
+      rec1 <- resources.resultsQueue.take
+      rec2 <- resources.resultsQueue.take
+      rec3 <- resources.resultsQueue.take
+      rec4 <- resources.resultsQueue.take
+      rec5 <- resources.resultsQueue.take
+    } yield Vector(rec1, rec2, rec3, rec4, rec5)
   } yield assert(
     gotAllRecords && resRecords
       .map(_.partitionKey())
       .diff(req.records().asScala.toVector.map(_.partitionKey()))
       .isEmpty,
     s"Got All Records: $gotAllRecords\nLength: ${resRecords.length}"
   )
-
+  
+  // For whatever reason, these tests seem to be flaky. Basically, a test will hit the timeout due to some
```

Comment:
For whatever reasons these tests are flaky now. I've tested this in both Jetty and Ember backends so I'm certain that this is an MUnit issue and not an Http4s issue (meaning the mock is fine). But I can't seem to nail down why this is happening.

I tried overriding things like the IORuntime and ExecutionContext for MUnit but still ran into issues. In fact I made things worse. 

CC @RaasAhsan